### PR TITLE
sysparam: buffer writes.

### PR DESCRIPTION
**Pull request by ourairquality**
If the source is in the flash then the write to the flash seems to just write zeros, and since it is common for the key to be a constant string, buffer all writes.

Perhaps someone could check if this makes sense?

### CODE REVIEW

File: core/sysparam.c [hunk 1]
```diff
@@ -116,16 +116,18 @@ static struct {
...
     return SYSPARAM_OK;
 }
 
 static inline IRAM sysparam_status_t _do_verify(uint32_t addr, const void *data, void *buffer, size_t len) {
     CHECK_FLASH_OP(sdk_spi_flash_read(addr, buffer, len));
-    if (memcmp(data, buffer, len)) {
+    if (memcmp(data, buffer, len))
         return SYSPARAM_ERR_IO;
-    }
+
```
Comment (foogod) [inline core/sysparam.c]: Why are you removing these braces?  Not having braces on multi-line `if` constructs is (IMHO) very bad coding practice and should be discouraged.

**Comment by foogod** [comment]
Aside from my comment about unnecessarily removing useful braces, this looks like a reasonable approach to solving the problem..

File: core/sysparam.c (hunk 1 above)
Comment (ourairquality) [inline core/sysparam.c]: Guess it all depends on your 'good coding style', sorry. Many coding styles omit braces if there is a single statement for the `if` and no `else`. Shall put it back!

Comment (kanflo) [inline core/sysparam.c]: IMHO (regardless of coding style) I prefer to have light weight patches only containing changes relevant for the PR. General code cleanup, fixing indentation, renaming variables for clarity I like to see in separate PRs.

**Comment by ourairquality** [comment]
Moved this along a lot further in https://github.com/SuperHouse/esp-open-rtos/pull/195

**Pull request closed without merging**